python_modules/dagster/dagster/_core/definitions/declarative_scheduling/operands/scheduled_since_condition.py

```diff
@@ -0,0 +1,119 @@
+
+    @property
+    def lookback_timedelta(self) -> datetime.timedelta:
+        return self.serializable_lookback_timedelta.to_timedelta()
+
+    @staticmethod
+    def from_lookback_delta(lookback_delta: datetime.timedelta) -> "ScheduledSinceCondition":
+        return ScheduledSinceCondition(
+            serializable_lookback_timedelta=SerializableTimeDelta.from_timedelta(lookback_delta)
+        )
+
+    def _get_minimum_timestamp(self, context: SchedulingContext) -> float:
+        """The minimum timestamp for a request to be considered in the lookback window."""
+        return (context.effective_dt - self.lookback_timedelta).timestamp()
+
+    def _get_new_slices_with_metadata(
+        self, context: SchedulingContext
+    ) -> Sequence[AssetSliceWithMetadata]:
+        """Updates the stored information as to when the asset was last requested."""
+        # the first time this asset has been evaluated
+        if context.previous_evaluation_info is None:
+            return []
+
+        previous_slices_with_metadata = (
+            context.previous_evaluation_node.slices_with_metadata
+            if context.previous_evaluation_node
+            else []
+        )
+
+        # no new updates since previous tick
+        if context.previous_requested_slice is None:
+            return previous_slices_with_metadata
+
+        # for existing subsets, remove references to newly-requested partitions, as these subsets
+        # are meant to represent the most recent time that the asset was requested
+        slices_with_metadata = [
+            AssetSliceWithMetadata(
+                asset_slice.compute_difference(context.previous_requested_slice), metadata
+            )
+            for asset_slice, metadata in previous_slices_with_metadata
+        ]
+
+        # for the newly-requested slice, add a new entry indicating that these partitions were
+        # requested on the previous tick
+        previous_request_timestamp = (
+            context.previous_evaluation_info.temporal_context.effective_dt.timestamp()
+        )
+        slices_with_metadata.append(
+            AssetSliceWithMetadata(
+                context.previous_requested_slice,
+                {_REQUEST_TIMESTAMP_METADATA_KEY: MetadataValue.float(previous_request_timestamp)},
+            )
+        )
+
+        # finally, evict any empty subsets from the list, and any subsets with an older timestamp
+        return [
+            asset_slice_with_metadata
+            for asset_slice_with_metadata in slices_with_metadata
+            if not (
+                asset_slice_with_metadata.asset_slice.is_empty
+                or cast(
+                    float,
+                    asset_slice_with_metadata.metadata.get(
+                        _REQUEST_TIMESTAMP_METADATA_KEY, MetadataValue.float(0)
+                    ).value,
+                )
+                < self._get_minimum_timestamp(context)
+            )
+        ]
```

Comment:
Similarly here this feels like it should be a primitive on a slightly higher level of abstraction.